src/auth/auth.service.ts

```diff
@@ -0,0 +1,45 @@
+import { Injectable, UnauthorizedException } from '@nestjs/common';
+import { JwtService } from '@nestjs/jwt';
+import * as bcrypt from 'bcrypt';
+
+import { UserService } from '@src/user/user.service';
+import { CreateUserDto } from '@src/user/dto/create-user.dto';
+import { LoginUserDto } from './dto';
+
+@Injectable()
+export class AuthService {
+  constructor(
+    private readonly userService: UserService,
+    private readonly jwtService: JwtService,
+  ) {}
+
+  async register(createUserDto: CreateUserDto) {
+    const user = await this.userService.create(createUserDto);
+
+    return { ...user, token: this.generateJwtToken(user.id) };
+  }
+
+  async login(loginDto: LoginUserDto) {
+    const { email, password } = loginDto;
+    const user = await this.userService.findOne(email);
+
+    if (!user) {
+      throw new UnauthorizedException('Invalid credentials');
+    }
+
+    if (!this.isValidPassword(password, user.password))
+      throw new UnauthorizedException('Invalid credentials');
```

Comment:
WDYT about creating a constant to keep all you error messages there?
Something like 
```
export const ErrorMessage = {
   INVALID_CREDENTIALS = 'Invalid credentials'
}
```

Reply:
done